pkg/pgmodel/upgrade_tests/upgrade_test.go

```diff
@@ -159,89 +112,48 @@ func TestUpgradeFromPrev(t *testing.T) {
 	//      gathered the db info in parallel. Unfortunately our db runner doesn't
 	//      support this yet
 	var upgradedDbInfo dbInfo
-	withNewDB(t, cleanImage, prevVersion.String(), *testDatabase,
-		func(container testcontainers.Container, connectURL string, db *pgxpool.Pool, tmpDir string) {
-			ingestor, err := pgmodel.NewPgxIngestor(db)
-			if err != nil {
-				t.Fatalf("error connecting to DB: %v", err)
-			}
-			_, err = ingestor.Ingest(copyMetrics(preUpgradeData1), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
-			_, err = ingestor.Ingest(copyMetrics(preUpgradeData2), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
-			ingestor.Close()
-			defer db.Close()
+	withDBAndConnector(t, *testDatabase, "timescale/timescale-prometheus:"+prevVersion.String(),
+		func(dbContainer testcontainers.Container, dbTmpDir string, connectorHost string, connectorPort nat.Port) {
+			client := http.Client{}
+			defer client.CloseIdleConnections()
+
+			writeUrl := fmt.Sprintf("http://%s:%s/write", connectorHost, connectorPort.Port())
+			t.Log(writeUrl)
+
+			doWrite(t, &client, writeUrl, preUpgradeData1, preUpgradeData2)
 
 			t.Logf("upgrading versions %v => %v", prevVersion, version.Version)
+			// TODO this should be NoSuperuser expect for the extwlist error
+			connectURL := testhelpers.PgConnectURL(*testDatabase, testhelpers.Superuser)
 			migrateToVersion(t, connectURL, version.Version, "azxtestcommit")
 
-			db, err = pgxpool.Connect(context.Background(), connectURL)
+			db, err := pgxpool.Connect(context.Background(), connectURL)
 			if err != nil {
 				t.Fatal(err)
 			}
 			defer db.Close()
-			ingestor, err = pgmodel.NewPgxIngestor(db)
+			ingestor, err := pgmodel.NewPgxIngestor(db)
 			if err != nil {
 				t.Fatalf("error connecting to DB: %v", err)
 			}
-			_ = ingestor.CompleteMetricCreation()
-			_, err = ingestor.Ingest(copyMetrics(postUpgradeData1), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
-			_, err = ingestor.Ingest(copyMetrics(postUpgradeData2), &prompb.WriteRequest{})
-			if err != nil {
-				t.Fatalf("ingest error: %v", err)
-			}
-			_ = ingestor.CompleteMetricCreation()
+
+			doIngest(t, ingestor, postUpgradeData1, postUpgradeData2)
+
 			ingestor.Close()
-			upgradedDbInfo = getDbInfo(t, container, tmpDir, db)
+			upgradedDbInfo = getDbInfo(t, dbContainer, dbTmpDir, db)
 		})
 
 	var pristineDbInfo dbInfo
-	withNewDB(t, cleanImage, version.Version, *testDatabase,
+	withNewDB(t, version.Version, *testDatabase,
```

Comment:
lets break this out into getPristineDbInfo() DbInfo

Reply:
what's the "this" here?